kopf/reactor/handling.py

```diff
@@ -240,8 +240,8 @@ class HandlerChildrenRetry(HandlerRetryError):
     # finalizers, lock it to this operator. Not all newly created objects will
     # produce an 'ACQUIRE' causation event. This only happens when there are
     # mandatory deletion handlers registered for the given object, i.e. if finalizers
-    # are required.
-    if cause.event == causation.ACQUIRE:
+    # are required. Only do this if there are handlers to execute.
+    if not skip and cause.event == causation.ACQUIRE:
```

Comment:
I think this condition cannot be True as for not `HANDLER_CAUSES` the `skip` variable is aways None.

Reply:
True. It could be `if handlers and ...` here.

Keep in mind that if a finalizer is not added because of this `if` here, the `detect_cause()` will believe that it should be added on the next iteration, and so on forever, until it is actually added. The cause handling will not happen until then.

The best solution here would be to **_not_** detect the `ACQUIRE` cause if there are no deletion handlers for that specific individual object.

Now, it is done above in the same file by a resource kind only: `requires_finalizer=registry.requires_finalizer(resource=resource)`. It should be the whole individual object, it seems, not just a resource kind. 

---

Keep in mind a tricky scenario:

* A non-matching object is created.
* No deletion handlers are selected by filters.
* Therefore, no finalizer is added.
* The handling cycle is done, the system is idling.
* Suddenly, the object gets its labels/annotations assigned, which makes it matching the filters.

Before starting an `UPDATE` cause handling, the `ACQUIRE` cause should be pre-executed, and the finalizer should be added (because the criteria for its existence have changed).